sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java

```diff
@@ -560,4 +560,25 @@ public static SharedKeyCredential getSharedKeyCredential(HttpPipeline httpPipeli
                 }
             });
     }
+
+    /**
+     * A utility method for converting the input stream to Flux of ByteBuffer. Will check InputSteam length
+     * by getting available()
+     *
+     * @param data The input data which needs to convert to ByteBuffer.
+     * @param blockSize The size of each ByteBuffer.
+     * @return {@link ByteBuffer} which contains the input data.
+     * @throws UnexpectedLengthException when input data length mismatch input length.
+     * @throws RuntimeException When I/O error occurs.
+     */
+    public static Flux<ByteBuffer> convertStreamToByteBuffer(InputStream data, int blockSize) {
+        final long length;
+        try {
+            length = data.available();
+        } catch (IOException e) {
+            throw LOGGER.logExceptionAsError(new RuntimeException("I/O errors occurs. Error deatils: "
+                + e.getMessage()));
+        }
+        return convertStreamToByteBuffer(data, length, blockSize);
```

Comment:
Shouldn't call into a method that is expecting the stream to have a known size, the whole point of the upload method being added is that is doesn't need to be known.

This call will almost always throw an exception